### PR TITLE
Fix Issue#18

### PR DESCRIPTION
**Pull request by Mukarr**
An [Esc] Hotkey binding was added to all the dialog boxes as required by Issue #18 .
File Editor, Graph editor and Dictionary dialog also close on [Esc], but this can be changed to minimize later.

### CODE REVIEW

File: src/main/java/fr/gramlab/frames/NewFileDialog.java [hunk 1]
```diff
@@ -56,45 +60,42 @@ public void actionPerformed(ActionEvent e) {
...
-	                       "You must indicate a file name!", "Error",
-	                       JOptionPane.ERROR_MESSAGE);
+					JOptionPane.showMessageDialog(null, "You must indicate a file name!", "Error",
+							JOptionPane.ERROR_MESSAGE);
 					return;
 				}
-				File f=new File(file,text.getText());
+				File f = new File(file, text.getText());
 				if (f.exists()) {
-					JOptionPane.showMessageDialog(null,
```
Comment (martinec) [inline src/main/java/fr/gramlab/frames/NewFileDialog.java]: Do not transform formatting of previous sources.

File: src/main/java/fr/gramlab/frames/NewFileDialog.java [hunk 2]
```diff
@@ -56,45 +60,42 @@ public void actionPerformed(ActionEvent e) {
...
-		                       "A file with that name already exists !", "Error",
-		                       JOptionPane.ERROR_MESSAGE);
+					JOptionPane.showMessageDialog(null, "A file with that name already exists !", "Error",
```
Comment (martinec) [inline src/main/java/fr/gramlab/frames/NewFileDialog.java]: Do not transform formatting of previous sources.

File: src/main/java/fr/gramlab/util/KeyUtil.java [hunk 3]
```diff
@@ -28,4 +31,61 @@ public void keyTyped(KeyEvent e) {
 	public static void addCRListener(JButton b) {
 		addCRListener(b,b);
 	}
+	
+	/**
```
Comment (martinec) [inline src/main/java/fr/gramlab/util/KeyUtil.java]: Try to use a javadoc-style comment

File: src/main/java/fr/gramlab/util/KeyUtil.java [hunk 4]
```diff
@@ -28,4 +31,61 @@ public void keyTyped(KeyEvent e) {
...
+	 * Added by Mukarram Tailor
+	 * 
+	 * Pressing Esc on a focused dialog will act as clicking cancel
+	 * 
+	 */
```
Comment (martinec) [inline src/main/java/fr/gramlab/util/KeyUtil.java]: Try to use a javadoc-style comment

File: unitex/src/fr/umlv/unitex/utils/KeyUtil.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
+package fr.umlv.unitex.utils;
```
Comment (martinec) [inline unitex/src/fr/umlv/unitex/utils/KeyUtil.java]: Each .java file must begin with the LGPL disclaimer.

**Comment by aleksandrachasch** [comment]
Hey @Mukarr,

While reviewing your PR, I imply two notions:
1) If a dialog has "Cancel" button that closes this dialog, [Esc] hotkey should close the dialog.
2) If a dialog has "Cross" button at top right corner that closes this dialog, [Esc] hotker should close the dialog.

According to these notions, while testing under Windows, 20 dialog boxes do not close if pressing [Esc]. This is how to obtain them:
0. Launch Unitex.jar
1. Select  Text -> Apply Lexical Resources
2. Select Text-> Locate Pattern
3. Select Text -> Apply CasSys Cascade
4. Select Text -> Located Sequences
5. Select Text -> Compile Elag Grammar
6. Select Text -> Construct FST-Text
7. Select Text -> Located Sequences
9. Select Text -> Convert FST-Text to Text...
10. Select Text -> Located Sequences
11. Select DELA -> Open -> dela-en-public.bin -> "Transcoding" dialog (as shown on screenshot)
transcoding_dialog
12. Select DELA -> Lookup...
13. Select DELA -> Transliterate
14. Select FSGraph -> New Cascade
15. Select Lexicon-Grammar -> Compile to GRF...
16. Select XAlign -> Open files...
17. Select File Edition -> Transcode Files
18. Select Info -> Preferences...
19. Select Text-> Preprocess Text...
21. Select Text -> Lemmatize

Also, if doing following steps:
 a) Select Text-> Open -> 80jours.txt
 b) Do you want to preprocess the text? Answer: <b>YES</b>
 c) Press <b>GO!</b>
 Three windows will appear: `TextFrame`, `Token List`, and `Word Lists` (as on screenshot). `Token List` and `Word Lists` do not close if pressing [Esc].
 screenshot1

**Comment by Mukarr** [comment]
Hi @aleksandrachasch ! I have added the hotkey functionality to all the above mentioned dialog boxes except the one obtained by:

>  1. Launch Unitex.jar

>  2. Select Text -> Apply CasSys Cascade .
As the cancel button of the CasSys Cascade dialog is not working, I think we should create a new issue for the same.

**Comment by martinec** [comment]
In the next scenario, pressing <kbd>Esc</kbd> does not work as expected :

**What steps will reproduce the problem?**

1. Launch Unitex.jar
1. Select English as working language
1. FSGraph > Open > Graphs > Preprocessing > Sentence > Sentence.grf
1. Press <kbd>Esc</kbd>

**What is the expected behavior?**

The graph window containing Sentence.grf is closed. All the menu items related to an active graph are **disabled** :

good

**What do you see instead?**

The graph window containing Sentence.grf is closed. All the menu items related to an active graph are **enabled** :

bad

**Comment by Mukarr** [comment]
Fixed the issue of window begin open in background, even after pressing Esc.
Also, added functionality to minimize on pressing [Esc] for Graph frame, Text Frame (both Editor and Corpus), word lists and token list.

**Comment by aleksandrachasch** [comment]
Everything works good now! What about adding minimize on [ESC] button to the Concordance output window?

1. Launch Unitex.jar
2. Text -> Open...
3. Do you want to preprocess the text? Answer <b>YES</b>
4. Text -> Locate Pattern
5. Choose "Regular Expression", print "A" (or whatever letter or word).
6. Located Sequences -> Build Concordance
screenshot1

**Comment by aleksandrachasch** [comment]
Great! Approved

**Comment by aleksandrachasch** [comment]
Approved! :+1:

**Comment by martinec** [comment]
In the next scenario, pressing <kbd>Esc</kbd> still does not work as expected :

**What steps will reproduce the problem?**

1. Launch Unitex.jar
1. Select English as working language
1. Text > Open > Corpus > ivanhoe.txt
1. Text > Construct FST-Text
1. Click Construct FST
1. On the 'FST-Text' window, press <kbd>Esc</kbd>
1. Similarly, on the 'Tfst tag list' window,  press <kbd>Esc</kbd>

**Comment by martinec** [comment]
LGTM. Great work! Could you please rebase on master and push again?

**Comment by martinec** [comment]
Thank you!

**Pull request merged**